galaxy_ng/app/access_control/access_policy.py

```diff
@@ -0,0 +1,92 @@
+from django.conf import settings
+from rest_access_policy import AccessPolicy
+
+from galaxy_ng.app import models
+from galaxy_ng.app import constants
+
+from galaxy_ng.app.access_control.statements import STANDALONE_STATEMENTS, INSIGHTS_STATEMENTS
+
+
+class AccessPolicyBase(AccessPolicy):
+    def get_policy_statements(self, request, view):
+        if (settings.GALAXY_DEPLOYMENT_MODE
+                == constants.DeploymentMode.INSIGHTS.value):
+
+            return INSIGHTS_STATEMENTS.get(self.NAME, [])
+        else:
+            return STANDALONE_STATEMENTS.get(self.NAME, [])
+
+    # used by insights access policy
+    def has_rh_entitlements(self, request, view, permission):
+        if not isinstance(request.auth, dict):
+            return False
+        header = request.auth.get('rh_identity')
+        if not header:
+            return False
+        entitlements = header.get('entitlements', {})
+        entitlement = entitlements.get(settings.RH_ENTITLEMENT_REQUIRED, {})
+        return entitlement.get('is_entitled', False)
+
+
+class NamespaceAccessPolicy(AccessPolicyBase):
+    NAME = 'NamespaceViewSet'
+
+
+class CollectionAccessPolicy(AccessPolicyBase):
+    NAME = 'CollectionViewSet'
+
+    def can_upload_to_namespace(self, request, view, permission):
+        namespace = None
+
+        # hack to check if view is CollectionUploadViewSet without creating
+        # circular imports
+        has_get_data = getattr(view, '_get_data', None)
```

Comment:
Please explain logic here.

Reply:
There are two viewsets that have to check if the user has permissions to modify a collection
- `CollectionUploadViewset`, which doesn't have a `get_object()` method like most viewsets do, because the object is being uploaded.
- `CollectionViewSet` which is used for updating deprecation status and functions like a normal viewset

`can_upload_to_namespace` has to support bot viewsets, and checking to see it `_get_data` is available is the best way to distinguish them. The initial implementation used `isinstance`, but that doesn't work because it creates circular imports.

Reply:
1. It sounds like there should be two policies for each view instead of one, having business logic specific for the view.
2. In case the first one is not possible you could use local imports (within the method) to avoid circular imports.

Reply:
I can split `can_upload_to_namespace` into `can_create_collection` and `can_update_collection`. That will avoid the `getattr` problem.

I don't want to create two access policies for collections, because that creates confusion about which access policies applies where